src/ArcgisAccount/useAccountManager.js

```diff
@@ -0,0 +1,179 @@
+  const name = accountManagerName
+    ? accountManagerName
+    : 'arcgis-account-manager';
+  const { accounts, status, active } = getAccountManagerStorage({ name });
+  const [accountManagerState, setAccountManagerState] = useState({
+    active: active,
+    accounts: accounts,
+    status: status
+  });
+  const authOptions = options;
+
+  /** Complete Login: If applicable */
+  useEffect(
+    () => {
+      const { loading, authProps } = status || {};
+      if (loading !== true) {
+        return;
+      }
+      //complete login
+      const { clientId, portalUrl, popup } = authProps;
+      completeAuth({ clientId, portalUrl, popup }).then(account => {
+        const { key, user } = account || {};
+        if (key) {
+          addAccountStorage({ name, key, account: user });
+        }
+
+        //Update localStorage/ state
+        completeStatusStorage({ name });
+        const accountManager = getAccountManagerStorage({ name });
+        setAccountManagerState(accountManager);
+      });
+    },
+    [accountManagerState]
+  );
+
+  /** Add Account */
+  const addAccount = useCallback(
+    options => {
+      const selectOptions = options ? options : authOptions;
+      const { clientId, redirectUri, portalUrl, popup } = selectOptions || {};
+
+      //set localstorage status
+      beginStatusStorage({ name, options: selectOptions });
+      //begin login
+      loginOAuth2({
+        clientId,
+        redirectUri,
+        portalUrl,
+        popup,
+        name,
+        setAccountManagerState
+      });
+    },
+    [accountManagerState]
+  );
+
+  /** Logout Account: Remove token and session from user auth object and attempt revoke token*/
+  const logoutAccount = useCallback(
+    key => {
+      const { accounts } = accountManagerState || {};
+      if (accounts[key]) {
+        const { session, token } = accounts[key];
+        const portalUrl = session ? session.portal : null;
+        const clientId = session ? session.clientId : null;
+        logoutOAuth2({
+          url: portalUrl,
+          clientId: clientId,
+          token: token
+        });
+
+        //Update localStorage/ state
+        logoutAccountStorage({ name, key });
+        const accountManager = getAccountManagerStorage({ name });
+        setAccountManagerState(accountManager);
+      }
+    },
+    [accountManagerState]
+  );
+
+  /** Remove Account: Remove account from local storage and attempt revoke token*/
+  const removeAccount = useCallback(
+    key => {
+      const { accounts } = accountManagerState || {};
+
+      if (accounts && accounts[key]) {
+        if (accounts[key].token) {
+          const { session, token } = accounts[key];
+          const portalUrl = session ? session.portal : null;
+          const clientId = session ? session.clientId : null;
+          //Remove token and session
+          logoutOAuth2({
+            url: portalUrl,
+            clientId: clientId,
+            token: token
+          });
```

Comment:
We could use Object Initializer shorthand here:

```suggestion
          logoutOAuth2({
            url: portalUrl,
            clientId,
            token
          });
```